src/services/pcn-packetcapture/src/Packetcapture.cpp

```diff
@@ -52,6 +56,14 @@ struct pcap_pkthdr {
 Packetcapture::Packetcapture(const std::string name, const PacketcaptureJsonObject &conf)
   : TransparentCube(conf.getBase(), { packetcapture_code_ingress }, { packetcapture_code_egress }),
     PacketcaptureBase(name) {
+  
+  gettimeofday(&ts, NULL);              //getting actual time
```

Comment:
Davide, one more comment: can you please add a couple of sentences here to present how the timestamping algorithm works? It is not immediate while looking at the code.

Reply:
Sure!
I'm doing it